Add LoginForm

shhhQuiettt opened the pull request:
Login form added
LoginForm test added
Protected route added
Code is formated now
Basic route added

shhhQuiettt committed:
Add LoginForm

Login form added
LoginForm test added
Protected route added
Code is formated now
Basic route added

shhhQuiettt committed:
fix ProtectedRoute

shhhQuiettt committed:
Add semicolons to end of lines

Add a tooon of semicolons

GarrusNapp commented on src/components/LoginForm.js:
```diff
@@ -0,0 +1,137 @@
+    const handleSubmit = (event) => {
+        event.preventDefault();
+
+        if (user.email == "" || user.pass == "") {
```
Widziałem, że macie potworzone funkcje walidacyjne. 
Niech one obsłużą puste hasła / emaile, tak, żeby cała walidacja odbywała się tam, a nie częściowo w tym komponencie.

GarrusNapp commented on src/components/LoginForm.js:
```diff
@@ -0,0 +1,137 @@
+                email = false;
+                setEmailValidError("To pole nie może być puste!");
+            }
+            if (user.pass === "") {
```
Ogólnie co do if, korzystajcie z tego że niektóre wartości są truthy/falsy np. `if (!user.pass)` zamiast `if (user.pass === "")`
Przy większej ilości ifów zwiększa to czytelność.

GarrusNapp commented on src/components/Input.js:
```diff
@@ -1,32 +1,69 @@
+                        className={`invalid-message invalid-message-${invMsgDirection}`}
+                        data-testid="invalid-msg"
+                        onMouseDown={(event) =>
+                            setIsValid({ ...isValid, [id]: true })
```
`[id]` to to samo co [`${id}`]

GarrusNapp commented on src/components/LoginForm.js:
```diff
@@ -0,0 +1,137 @@
+import Input from "./Input";
+
+const LoginForm = ({ login }) => {
+    const [user, setUser] = useState({
```
Zamiast dwóch state `user` i `isValid` zrobiłbym 4 dla każdego pola. Będzie wtedy można pozbyć się obsługiwania `id` w inpucie, co uprości kod. 

Podobnie można uprościć obsługę errorów i zamiast używać booleanu valid, to dać array z errorami i w zależności czy jest pusty czy nie dodawać odpowiednie klasy i message.

GarrusNapp commented on src/components/LoginForm.js:
```diff
@@ -0,0 +1,137 @@
+            const err = login(user);
+            if (err !== null) {
+                switch (err) {
+                    case "incorret password or email": {
```
*incorrect

GarrusNapp commented on src/components/Input.js:
```diff
@@ -1,32 +1,69 @@
+                placeholder={placeholder}
+                value={value === undefined ? user[`${id}`] : value}
+                onChange={
+                    onChange == undefined
```
jak nie przyszedł prop onChange to nie robiłbym domyślnie żeby obsługiwał setUser, bo nie będzie się dało użyć gdzie indziej inputu.

GarrusNapp commented on src/service.js:
```diff
@@ -1,85 +1,133 @@
+        .then((response) => {
+            const obj = JSON.parse(response);
+            if (obj.success) {
+                isAuthenticated = true;
```
authenticated = true*

GarrusNapp commented on src/service.js:
```diff
@@ -1,85 +1,133 @@
+            case "street":
+            case "city": {
+                const pattern =
+                    /^[a-z\u00AA\u00B5\u00BA\u00C0-\u00D6\u00D8-\u00F6\u00F8-\u02C1\u02C6-\u02D1\u02E0-\u02E4\u02EC\u02EE\u0370-\u0374\u0376\u0377\u037A-\u037D\u037F\u0386\u0388-\u038A\u038C\u038E-\u03A1\u03A3-\u03F5\u03F7-\u0481\u048A-\u052F\u0531-\u0556\u0559\u0561-\u0587\u05D0-\u05EA\u05F0-\u05F2\u0620-\u064A\u066E\u066F\u0671-\u06D3\u06D5\u06E5\u06E6\u06EE\u06EF\u06FA-\u06FC\u06FF\u0710\u0712 ...
```
co to za szalony pattern?

GarrusNapp commented on src/App.js:
```diff
@@ -1,18 +1,31 @@
-    </div>
-  );
+    return (
+        <>
```
Można też zrobić też standardowego switcha, tego typu:
```
        <Switch>
          <Route path="/about">
            <About />
          </Route>
          <Route path="/users">
            <Users />
          </Route>
          <Route path="/">
            <Home />
          </Route>
        </Switch>
```
funkcje createUser / login mogą być importowane już w komponentach, które ich używają.

GarrusNapp commented on src/components/Input.js:
```diff
@@ -1,32 +1,69 @@
+                    isValid[`${id}`] ? "" : "invalid-input"
+                }`}
+                placeholder={placeholder}
+                value={value === undefined ? user[`${id}`] : value}
```
value bym zawsze raczej używał tego z propsów, a nie wysyłał usera w propsach wcale.

shhhQuiettt committed:
Register/LoginForm and react-hook-forms are now happily married

Forms are react-hook-form now
service.test.js added
validation is executed by validation.js dependency
